src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -1257,15 +1262,18 @@ void HierRTLMP::runHierarchicalMacroPlacement(Cluster* parent)
     if (cluster->getClusterType() == StdCellCluster) {
       continue;
     }
-    Rect fence(-1.0, -1.0, -1.0, -1.0);
-    Rect guide(-1.0, -1.0, -1.0, -1.0);
+    Rect fence, guide;
+    fence.mergeInit();
+    guide.mergeInit();
```

Comment:
If the guide remains in the init state with no merges, does the subsequent relocate call confuse isValid?

Reply:
Relocate checks if the rect is valid before relocating, so if no fence/guide was merged, it will do nothing.